### PR TITLE
Post about flow identification in SnabbWall

### PR DESCRIPTION
**Pull request by aperezdc**

### CODE REVIEW

**Comment by aperezdc** [comment]
@dpino Could you PTAL?

File: 2016/identifying-l7-flows.markdown [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+In SnabbWall, the assignment of packets to flows is done based on the
+following fields from the packet:
+
+* [802.1Q](https://en.wikipedia.org/wiki/802.1q) VLAN tag.
+* Source and destination IP addresses.
+* Source and destination port numbers.
+
+The VLAN tag is there to force classifying packets with the same source and
+destination *but* in different logical networks in in separate packet flows.
```
Comment (dpino) [inline 2016/identifying-l7-flows.markdown]: s/in in/in/

File: 2016/identifying-l7-flows.markdown [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+As for port numbers, in practice these are only extracted from packets when
+the upper layer layer protocol is UDP or TCP. There are other protocols which
```
Comment (dpino) [inline 2016/identifying-l7-flows.markdown]: s/layer layer/layer/

File: 2016/identifying-l7-flows.markdown [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+addresses, ports) and calculates a unique *flow key* from them, while
+backend-specific code inspects the contents of the packet and identifies the
+application for a flow of packets. Once the generic part of the code has
+calculated a key, it can be used to keep tables which associate additional
+data to each flow. While SnabbWall has only one backend which at the moment
+which uses [nDPI][ndpi], this split makes it easier to add others in the
+future.
+
+For efficiency —both in terms of memory and CPU usage— flow keys are
+represented using a C `struct`. The following snippet shows the one for for
```
Comment (dpino) [inline 2016/identifying-l7-flows.markdown]: s/for for/for/

**Pull request closed without merging**